### PR TITLE
All log retention syncronized across log groups

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

Setting log retention across all log groups. Prod will retain indefinitely, other envs, 1 year.

# Test instructions | Instructions pour tester la modification

Tested apply in dev
Test in staging
Smoke test in staging
Verify log retention periods in AWS console.

### CODE REVIEW

File: aws/common/cloudwatch_log.tf [hunk 1]
```diff
@@ -4,6 +4,7 @@
 resource "aws_cloudwatch_log_group" "sns_deliveries" {
   count = var.cloudwatch_enabled ? 1 : 0
   name  = "sns/${var.region}/${var.account_id}/DirectPublishToPhoneNumber"
+  retention_in_days = var.env == "production" ? 0 : 365
```
Comment (jimleroyer) [inline aws/common/cloudwatch_log.tf:7]: no sns retention for staging?

File: aws/eks/cloudwatch_log.tf [hunk 2]
```diff
@@ -5,19 +5,19 @@
...
   count             = var.cloudwatch_enabled ? 1 : 0
   name              = "/aws/eks/${var.eks_cluster_name}/cluster"
-  retention_in_days = 14
+  retention_in_days = var.env == "production" ? 0 : 365
 }
 
 resource "aws_cloudwatch_log_group" "notification-canada-ca-eks-application-logs" {
   count             = var.cloudwatch_enabled ? 1 : 0
   name              = "/aws/containerinsights/${var.eks_cluster_name}/application"
-  retention_in_days = var.env == "production" ? 0 : 30
```
Comment (jimleroyer) [inline aws/eks/cloudwatch_log.tf:14]: 30 days for staging?

File: aws/common/cloudwatch_log.tf (hunk 1 above)
Comment (ben851) [reply aws/common/cloudwatch_log.tf:7]: 0 = indefinite, so if prod, no expiry, else 365 days

File: aws/eks/cloudwatch_log.tf (hunk 2 above)
Comment (ben851) [reply aws/eks/cloudwatch_log.tf:14]: welp that should've been 365

File: aws/common/cloudwatch_log.tf (hunk 1 above)
Comment (jimleroyer) [reply aws/common/cloudwatch_log.tf:7]: makes a lot more sense then, I forgot about that 0 value

File: aws/eks/cloudwatch_log.tf (hunk 2 above)
Comment (jimleroyer) [reply aws/eks/cloudwatch_log.tf:14]: perfect!

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**